packages/p2p/src/components/buy-sell/buy-sell-form.jsx

```diff
@@ -86,6 +101,19 @@ const BuySellForm = props => {
         }
     };
 
+    const onSubmitAdTransaction = async (...args) => {
+        const updated_market_rate = await fetchCurrentMarketRate();
```

Comment:
I believe you can subscribe to `exchange_rates` api.

Reply:
Yes. But we need to display a market rate change modal if the market rate has changed when the user had started to transact on the ad and when he has submitted the details. Hence the API is being called twice to keep track of change of values